clientconn.go

```diff
@@ -216,7 +216,14 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 	defer func() {
 		select {
 		case <-ctx.Done():
-			conn, err = nil, ctx.Err()
+			switch {
+			case ctx.Err() == err:
+				conn = nil
+			case err == nil || !cc.dopts.returnLastError:
+				conn, err = nil, ctx.Err()
+			default:
+				conn, err = nil, fmt.Errorf("%v: %v", ctx.Err(), err)
```

Comment:
Wouldn't it be better for the second `%v` to be `%w` here, so the error is wrapped and thus available for introspection?

Reply:
#4155 / #4163